### PR TITLE
Delete Message in Bulk

### PR DESCRIPTION
**Pull request by SinfulPhantom**
## Summary

Adds logic to utilize Revolt's bulk message call for #91 
https://developers.revolt.chat/api#tag/Messaging/operation/message_bulk_delete_req

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, …)

### CODE REVIEW

**Comment by SinfulPhantom** [comment]
@EnokiUN - This is my first time contributing to this project. I have created a solution, but I'm not exactly sure how I can test this locally. Any advice would be much appreciated.
Thanks in advance!

**Comment by SinfulPhantom** [comment]
I'm _**Squadwarden**_ in Revolt, if you need to reach me for anything

**Comment by EnokiUN** [comment]
> @EnokiUN - This is my first time contributing to this project. I have created a solution, but I'm not exactly sure how I can test this locally. Any advice would be much appreciated.
> Thanks in advance!

You can try creating a `test.py` in the root of your clone, it should be ignored bt the `.gitignore` and importing voltage there should use your own code

File: voltage/internals/http.py [hunk 1]
```diff
@@ -536,6 +536,19 @@ async def delete_message(self, channel_id: str, message_id: str):
...
+        Deletes messages in bulk
+
+        Parameters
+        ----------
+        channel_id: :class:`str`
+            The id of the channel.
+        message_ids: :class:`List[str]`
+            The id of the message.
+        """
+        return await self.request("DELETE", f"channels/{channel_id}/messages/{message_ids}")
```
Comment (EnokiUN) [inline voltage/internals/http.py:550]: According to the docs you pass the message IDs as a JSON object in the body of the request

File: voltage/messageable.py [hunk 2]
```diff
@@ -280,15 +280,23 @@ async def purge(self, amount: int):
...
-                    raise
+        message_list = []
+        [
+            message_list.append(message["_id"])
+            for message in await self.cache.http.fetch_messages(
+                channel_id, "Latest", limit=amount
+            )
+        ]
+        try:
+            await self.cache.http.bulk_delete_messages(
```
Comment (EnokiUN) [inline voltage/messageable.py:291]: you should probably chunk this as to not try to clear more than 100 messages at once

**Review by EnokiUN — CHANGES_REQUESTED** [review]

File: voltage/messageable.py (hunk 2 above)
Comment (SinfulPhantom) [reply voltage/messageable.py:292]: Changed in most recent commit

File: voltage/internals/http.py (hunk 1 above)
Comment (SinfulPhantom) [reply voltage/internals/http.py]: Utilizing json.dumps() for this now

Comment (EnokiUN) [reply voltage/internals/http.py]: that won't work, you have to pass it into the body of the request

**Comment by EnokiUN** [comment]
LGTM, just test it out and it'll be ready for merging. I sadly cannot currently test it myself since I'm abroad on vacation

**Comment by EnokiUN** [comment]
Any updates on this?

**Comment by SinfulPhantom** [comment]
This is something I put in the back burner and never got around to. Will try to get it tested soon I hope. Holidays are coming up, so I'll need to find time.

**Pull request closed without merging**